### DISCUSSION TITLE
Improvements on concealing

### DISCUSSION THREAD

**Discussion by voyeg3r** [jakewvincent/mkdnflow.nvim#66] [Ideas]
The plugin: https://github.com/prurigro/vim-markdown-concealed has nice concealing features, I think would be great to add some of them.

**Comment by jakewvincent** [comment]
Hi, thanks for the idea! I understand that concealing can be very nice (I myself enjoy a little bit of concealing, especially for links), but I'm unsure about adding more concealing features to the plugin since the "mission" of the plugin is to enhance maintenance and navigation of notes--not necessarily to make notes look nicer. It feels a little bit outside of the scope of the plugin, and I'm worried of trying to pack too much into a single plugin.

Here are some thoughts:
* On that last point (_...single plugin..._), one idea I'm considering is creating a separate plugin for conceal features that could be configured in a familiar way (passing a config to a setup function, rather than using vim variables and the like)
* On the other hand, I don't think there's any real benefit to concealing via Lua instead of vimscript (except for the above point about configuration) since Neovim doesn't have native Lua functions for adding matches for concealing/highlights. So part of me wants to say that people should just use the plugin you mentioned if they want nice conceal features (https://github.com/prurigro/vim-markdown-concealed). However, I see that that plugin has some feature overlap with mkdnflow, which means it wouldn't be ideal to combine this plugin with that one.
* The most recent markdown parsers for treesitter have built-in concealing (if markdown and markdown_inline are already installed, all that's needed is `set conceallevel=2` to enable the concealing). But I'm sure there are plenty of folks who still don't use treesitter, so perhaps this doesn't matter.

May I ask which conceal features from the plugin you linked you find most useful?
